docs/flow_filtering.md

```diff
@@ -0,0 +1,95 @@
+ 
+### Mandatory parameters
+
+- `FLOW_FILTER_IP_CIDR` - IP address and CIDR mask for the flow filter rule, supports IPv4 and IPv6 address format.
+  If wanted to match against any IP, user can use `0.0.0.0/0` or `::/0` for IPv4 and IPv6 respectively.
+- `FLOW_FILTER_ACTION` - Action to be taken for the flow filter rule. Possible values are `Accept` and `Reject`.
+  - For the matching rule with `Accept` action this flow will be allowed to be cached in eBPF table, with updated global metric `FlowFilterAcceptCounter`.
+  - For the matching rule with `Reject` action this flow will be dropped and not cached in eBPF table, with updated global metric `FlowFilterRejectCounter`.
+  - If the rule is not matched, the flow will be allowed to be cached in eBPF table, with updated global metric `FlowFilterNoMatchCounter`.
+
+### Optional parameters
+
+- `FLOW_FILTER_DIRECTION` - Direction of the flow filter rule. Possible values are `Ingress` and `Egress`.
+- `FLOW_FILTER_PROTOCOL` - Protocol of the flow filter rule. Possible values are `TCP`, `UDP`, `SCTP`, `ICMP`, `ICMPv6`.
+- `FLOW_FILTER_SOURCE_PORT` - Single Source port of the flow filter rule.
+- `FLOW_FILTER_SOURCE_PORT_RANGE` - Source port range of the flow filter rule. using "80-100" format.
+- `FLOW_FILTER_DESTINATION_PORT` - Single Destination port of the flow filter rule.
+- `FLOW_FILTER_DESTINATION_PORT_RANGE` - Destination port range of the flow filter rule. using "80-100" format.
+- `FLOW_FILTER_ICMP_TYPE` - ICMP type of the flow filter rule.
+- `FLOW_FILTER_ICMP_CODE` - ICMP code of the flow filter rule.
+- `FLOW_FILTER_IP` - Specific IP address of the flow filter rule.
+
+## How does it work
+
+### What does IP CIDR filtering work? 
+
+- When filtering ingress traffic, the source IP address is a crucial parameter used to match filter rules configured with CIDR notation.
+  CIDR notation allows for the efficient representation of IP address ranges by combining the base IP address with a prefix length
+  to denote the network subnet.
+- When filtering egress traffic, the destination IP address is a crucial parameter used to match filter rules configured with CIDR notation.
+  CIDR notation allows for the efficient representation of IP address ranges by combining the base IP address with a prefix length.
+
+### Matching specific endpoints using `FLOW_FILTER_IP`
+
+- When filtering ingress traffic, the `FLOW_FILTER_IP` will be used to match against the destination IP address of the packet, 
+  so after matching source IP CIDR now we can pinpoint to a specific endpoint via `FLOW_FILTER_IP` and based on the provisioned action,
+  flow with either be allowed to be cached in eBPF flow table or not.
+- When filtering egress traffic, the `FLOW_FILTER_IP` will be used to match against the source IP address of the packet,
+  so after matching destination IP CIDR now we can pinpoint to a specific endpoint via `FLOW_FILTER_IP` and based on the provisioned action,
```

Comment:
would it be correct to say `FLOW_FILTER_IP` will match the in-cluster IP address whether it's ingress or egress?

Reply:
yes